sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1325,3 +1388,48 @@ def test_pairwise_boolean_distance():
     nn1 = NN(metric="jaccard", algorithm='brute').fit(X)
     nn2 = NN(metric="jaccard", algorithm='ball_tree').fit(X)
     assert_array_equal(nn1.kneighbors(X)[0], nn2.kneighbors(X)[0])
+
+
+# If bonus > 0, we precompute more neighbors than necessary
+@pytest.mark.parametrize('bonus', [0, 2])
+def test_pipeline_with_nearest_neighbors_transformer(bonus):
+    # Test chaining KNeighborsTransformer and classifiers/regressors
+    rng = np.random.RandomState(0)
+    X = 2 * rng.rand(40, 5) - 1
+    X2 = 2 * rng.rand(40, 5) - 1
+    y = ((X ** 2).sum(axis=1) < .5).astype(np.int)
+
+    # K Neighbors
+    for klass in [neighbors.KNeighborsClassifier,
```

Comment:
I am not convinced you need to test both KNeighborsClassifier and *Regressor, but you do need to check that `RadiusNeighbors` can follow `KNeighborsTransformer` and vice-versa.